manual/sphinx/developer_docs/code_generation.rst

```diff
@@ -0,0 +1,40 @@
+.. _sec-fieldops:
+
+Field2D/Field3D Arithmetic Operators
+------------------------------------
+
+The arithmetic operators (`+`, `-`, `/`, `*`) for `Field2D` and
+`Field3D` are generated automatically using the `Jinja`_ templating
+system. This requires Python.
+
+Because this is fairly low-level code, and we don't expect it to change
+very much, the generated code is kept in the git repository. This has
+the benefit that Python and Jinja are not needed to build BOUT++, only
+to change the `Field` operator code.
+
+.. note:: You should not modify the generated code directly. Instead,
+          modify the template and re-generate the code.
+
+The Jinja template is in `src/field/gen_fieldops.jinja`, and the driver
+is `src/field/gen_fieldops.py`. The driver loops over every combination
+of `BoutReal`, `Field2D`, `Field3D` (collectively just "fields" here)
+with the arithmetic operators, and uses the template to generate the
+appropriate code. There is some logic in the template to handle certain
+combinations of the input fields: for example, for the binary infix
+operators, only check the two arguments are on identical meshes if
+neither is `BoutReal`.
+
+To install Jinja:
+
+.. code-block:: console
+
+   $ pip install --user Jinja2
```

Comment:
Presumably this should be `pip3` on systems where the default python version is not 3? The gen_fieldops.py specifically requests v3. It may be worth noting this.